op-service/sources/batching/test/event_stub.go

```diff
@@ -0,0 +1,63 @@
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-service/sources/batching/rpcblock"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+type expectedFilterLogsCall struct {
+	topics  [][]common.Hash
+	to      common.Address
+	block   rpcblock.Block
+	outputs []types.Log
+	err     error
+}
+
+func (c *expectedFilterLogsCall) Matches(rpcMethod string, args ...interface{}) error {
+	if rpcMethod != "eth_getFilterLogs" {
+		return fmt.Errorf("expected rpcMethod eth_getFilterLogs but was %v", rpcMethod)
+	}
+
+	topics := args[0].([][]common.Hash)
+
+	if !reflect.DeepEqual(topics, c.topics) {
+		return fmt.Errorf("expected topics %v but was %v", c.topics, topics)
+	}
+
+	to := args[1].(common.Address)
+	if to != c.to {
+		return fmt.Errorf("expected contract address %v but was %v", c.to, to)
+	}
+	return c.err
+}
+
+func (c *expectedFilterLogsCall) Execute(t *testing.T, out interface{}) error {
+	j, err := json.Marshal(c.outputs)
+	require.NoError(t, err)
+	json.Unmarshal(j, out)
```

Comment:
should check `require.NoError(t, json.Unmarshal(j, out))` ?

Reply:
good point. 
Done in https://github.com/ethstorage/optimism/pull/63/commits/94a61338fb885628c390d579d14413d18d7556ed